app/src/main/java/com/sunasterisk/youtubebackground/screen/main/MainActivity.kt

```diff
@@ -2,12 +2,77 @@ package com.sunasterisk.youtubebackground.screen.main
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentTransaction
 import com.sunasterisk.youtubebackground.R
+import com.sunasterisk.youtubebackground.screen.fragment.colectiontrending.ColectionTrendingFragment
+import com.sunasterisk.youtubebackground.screen.fragment.home.HomeFragment
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+    val fragmentManager = supportFragmentManager
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        setSupportActionBar(toolbarMain)
+        addFirstFragment()
+    }
+
+    private fun addFirstFragment() {
+        fragmentManager.beginTransaction()
+            .add(R.id.frameContent, HomeFragment.newInstance())
+            .addToBackStack(null)
+            .commit()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.top_bar_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when(item.itemId) {
+            R.id.iconMenu -> {
+                iconMenuOnClick()
+            }
+        }
+        return true
+    }
+
+    private fun iconMenuOnClick() {
+        if (linearSideMenu.visibility == View.GONE) {
+            linearSideMenu.visibility = View.VISIBLE
+        } else {
+            linearSideMenu.visibility = View.GONE
+        }
+    }
+
+    override fun onBackPressed() {
+        if (fragmentManager.backStackEntryCount > 1){
+            fragmentManager.popBackStack()
+        }
+    }
+
+    public fun replaceFragment(view: View) {
+
+        var fragmentTransaction = fragmentManager.beginTransaction()
+        var fragment: Fragment? = null
+        when(view.id) {
+            R.id.buttonHome -> fragment = HomeFragment.newInstance()
+            R.id.buttonTrending -> fragment = ColectionTrendingFragment.newInstance()
+        }
+
+        if (fragment != null) {
+            fragmentTransaction.replace(R.id.frameContent, fragment)
+        }
+
+        fragmentTransaction.addToBackStack(null)
+        fragmentTransaction.commit()
```

Comment:
nếu fragment null cũng commit sao 
```suggestion
        var fragment = when(view.id) {
            R.id.buttonHome -> fragment = HomeFragment.newInstance()
            R.id.buttonTrending -> fragment = ColectionTrendingFragment.newInstance()
            else -> null 
        }
```

Reply:
đây là hàm của button anh! nếu click button là hàm chạy ! mà đã click button là nó chắc chắn có fragment